docs/src/tutorial.md

```diff
@@ -0,0 +1,157 @@
+
+## Usage Examples
+
+### First 1D Example
+
+Suppose we want to compute the range of the function
+
+```math
+f(x) = -\sum_{k=1}^5kx\sin\left(\frac{k(x-3)}{3}\right)
+```
+
+over the domain ``X = [-10, 10]``.
+
+If we call `enclose` without specifying the algorithm, it will evaluate ``f(X)`` using plain interval arithmetic (this is called *natural enclosure*), as the following example shows.
+
+```@example tutorial
+f(x) = -sum(k*x*sin(k*(x-3)/3) for k in 1:5)
+X = -10..10
+Y = enclose(f, X)
+```
+
+Generally, using natural enclosures leads to unpleasantly large overestimates, this is generally due to the [dependency problem](https://en.wikipedia.org/wiki/Interval_arithmetic#Dependency_problem). To overcome this, you probably want to use some other methods in your application. The next example bounds the range using the [`BranchAndBoundEnclosure`](@ref) algorithm.
+
+```@example tutorial
+Ybb = enclose(f, X, BranchAndBoundEnclosure())
+```
+
+As you can see, the result is much tighter now, while still being rigorous! The results can be visualized using e.g. `Plots.jl`.
+
+```@example tutorial
+using Plots
+plot(IntervalBox(X, Y), label="natural enclosure")
+plot!(IntervalBox(X, Ybb), label="Branch and Bound")
+plot!(f, -10, 10, lw=2, c=:black, label="f")
+savefig("ex1.png") # hide
+```
+
+![](ex1.png)
+
+### Tuning parameters
+
+Some algorithms have parameters that can be tuned. For example looking at [`BranchAndBoundEnclosure`](@ref) documentation, we can see that it has two parameters : `tol` and `maxdepth`.
+If you want to use different values from the default ones, you can pass the parameters as keyword arguments to the algorithm constructor. For example, you can limit the depth of the search tree to `6` the following way
+
+```@example tutorial
+enclose(f, X, BranchAndBoundEnclosure(; maxdepth=6))
+```
+
+Generally, tuning parameters can be a good idea to achieve the desired accuracy-tradeoff in your application.
+
+### Combining different methods
+
+Sometimes there is no "best" algorithm, one might give a tighter estimate of the range upper bound and another a tighter estimate on the lower bound. In this case, the result can
+be achieved by taking the intersection of both results. For example, let us consider the function ``g(x) = x^2 - 2x + 1``, which we want to evaluate over the domain ``X_g = [0, 4]``. Let us first try using plain interval arithmetic
+
+```@example tutorial
+g(x) = x^2 - 2*x + 1
+Xg = 0..4
+enclose(g, Xg, NaturalEnclosure()) # this is equivalent to enclose(g, Xg)
+```
+
+Now let us bound the range using [`MeanValueEnclosure`](@ref), which uses the mean-value form of the function.
+
+```@example tutorial
+enclose(g, Xg, MeanValueEnclosure())
+```
+
+As you can see, there is no clear winner and a better enclosure could be obtained by taking the intersection of the two results. This can be easily done in one pass by passing to `enclose` a vector of methods
```

Comment:
```suggestion
As you can see, there is no clear winner and a better enclosure could be obtained by taking the intersection of the two results. This can be easily done in one command by passing a vector of methods to `enclose`:
```